wcfsetup/install/files/js/WCF.Attachment.js

```diff
@@ -285,20 +292,112 @@ if (COMPILER_TARGET_DEFAULT) {
 		 * @see        WCF.Upload._upload()
 		 */
 		_upload: function (event, file, blob) {
-			var $uploadID = undefined;
+			var _super = this._super.bind(this);
 			
-			if (this._validateLimit()) {
-				$uploadID = this._super(event, file, blob);
-			}
-			
-			if (this._fileUpload) {
-				// remove and re-create the upload button since the 'files' property
-				// of the input field is readonly thus it can't be reset
-				this._removeButton();
-				this._createButton();
-			}
-			
-			return $uploadID;
+			require([
+				'WoltLabSuite/Core/FileUtil',
+				'WoltLabSuite/Core/ImageResizer',
+				'WoltLabSuite/Core/Ajax/Status'
+			]).then((function (modules) {
+				var FileUtil = modules[0];
+				var ImageResizer = modules[1];
+				var AjaxStatus = modules[2];
+				
+				AjaxStatus.show();
+				
+				var $files = [];
+				
+				if (file) {
+					$files.push(file);
+				}
+				else if (blob) {
+					$files.push(FileUtil.blobToFile(blob, 'pasted-from-clipboard'));
+				}
+				else {
+					$files = this._fileUpload.prop('files');
+				}
+				
+				// Default action is to resolve with the unaltered list of files
```

Comment:
In general: Comments should be a whole sentence ending with a period and they should describe _why_ something is done, not _how_.